source/app/service-providers/appearance/index.ts

```diff
@@ -13,39 +13,45 @@
  * END HEADER
  */
 
-import EventEmitter from 'events'
 import {
   ipcMain,
   nativeTheme,
   systemPreferences
 } from 'electron'
+import LogProvider from '../log'
+import ProviderContract from '../provider-contract'
 
 /**
  * This class manages automatic changes in the appearance of the app. It won't
  * do anything, if the scheduler is set to off, listen to changes in the
  * operating system's appearance if set to system, and switch the mode on a given
  * time if set to schedule.
  */
-export default class AppearanceProvider extends EventEmitter {
+export default class AppearanceProvider extends ProviderContract {
   private _mode: 'off'|'system'|'schedule'|'auto'
   private _scheduleWasDark: boolean
   private _startHour: number
   private _startMin: number
   private _endHour: number
   private _endMin: number
+
+  private readonly _logger: LogProvider
+  private readonly _config: ConfigProvider
   /**
    * Create the instance on program start and initially load the settings.
    */
-  constructor () {
+  constructor (logger: LogProvider, config: ConfigProvider) {
```

Comment:
Or shorter:
```suggestion
  constructor (private readonly _logger: LogProvider, private readonly _config: ConfigProvider) {
```
without the explicit class attributes `_logger` and `_config`.

Reply:
Nice find, thanks!